config/config_test.go

```diff
@@ -97,3 +99,76 @@ func TestAFlagAndBFlag(t *testing.T) {
 		})
 	}
 }
+
+func TestInitJSONConfig(t *testing.T) {
+	jsonConfig := `{
+		"server_address": "localhost:8080",
+		"base_url": "http://localhost:8080",
+		"file_storage_path": "/tmp/storage",
+		"database_dsn": "/dbname",
+		"enable_https": true
+	}`
+	jsonFile, err := os.CreateTemp("", "config.json")
+	assert.NoError(t, err)
+	defer os.Remove(jsonFile.Name())
+
+	_, err = jsonFile.WriteString(jsonConfig)
+	assert.NoError(t, err)
+	err = jsonFile.Close()
+	assert.NoError(t, err)
+
+	var config Config
+
+	config.Config = jsonFile.Name()
+	err = config.InitJSONConfig()
+	if err != nil {
+		return
+	}
+	assert.Equal(t, "localhost:8080", config.ServerURL)
+	assert.Equal(t, "http://localhost:8080", config.BaseShortURL)
+	assert.Equal(t, "/tmp/storage", config.FileStoragePath)
+	assert.Equal(t, "/dbname", config.DataBaseDsn)
+	assert.True(t, config.EnableHTTPS)
+	assert.Equal(t, jsonFile.Name(), config.Config)
+}
```

Comment:
Кажется забыл сделать assert  ошибки. Так же можно удобно сравнивать всю структуру при помощи `cmp.Diff(...)`. Это поможет более наглядно увидеть разницу в структурах в виде патча.
```suggestion
	assert.NoError(t, err)

	wantConfig := Config{
		ServerURL:       "localhost:8080",
		BaseShortURL:    "http://localhost:8080",
		FileStoragePath: "/tmp/storage",
		DataBaseDsn:     "/dbname",
		EnableHTTPS:     true,
		Config:          jsonFile.Name(),
	}

	if diff := cmp.Diff(wantConfig, config); diff != "" {
		t.Errorf("Config mismatch (-expected +got):\n%s", diff)
	}
```